inc/functions/files.php

```diff
@@ -839,12 +855,15 @@ function rocket_clean_domain( $lang = '' ) {
 		*/
 		do_action( 'before_rocket_clean_domain', $root, $lang, $url ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals
 
-		// Delete cache domain files.
-		$dirs = glob( $root . '*', GLOB_NOSORT );
-		if ( $dirs ) {
-			foreach ( $dirs as $dir ) {
-				rocket_rrmdir( $dir, get_rocket_i18n_to_preserve( $lang ) );
+		try {
+			$files = new RegexIterator( $iterator, "/{$file['host']}*{$file['path']}/i" );
+
+			foreach ( $files as $file ) {
+				rocket_rrmdir( $file, get_rocket_i18n_to_preserve( $lang ) );
```

Comment:
We could do this before the first `foreach` loop on L837.
```php
$dirs_to_preserve = get_rocket_i18n_to_preserve( $lang );
```
And maybe also this, while at it:
```php
$url_no_dots = apply_filters( 'rocket_url_no_dots', false );
```